easycukes-rest/src/main/java/com/worldline/easycukes/rest/stepdefs/RESTStepdefs.java

```diff
@@ -448,4 +447,24 @@ public void checkIfParameterEquals(String parameter, String value)
     public void setDateFormat(String format) throws Throwable {
         CukesHelper.setParameter(RestConstants.DATE_FORMAT, format);
     }
+    
+    @Given("^I set the parameter \"(.*?)\" to first not null property \"(.*?)\" in response array$")
+	public void setAParameterToAFirstNotNullPropertyFromResponseArray(final String param, final String property)
+			throws ParseException {
+		CukesHelper.setParameter(param, RestService.getInstance().getIdFromResponseArrayByProperty(property));
+	}
+
+
+	@When("^I send put to \"(.*?)\" the parameters:$")
```

Comment:
Yes we have "^I (post|put|delete) to \"([^\"]*)\" the parameters:$" but here proxy issue is not handled. 
I found "^I send post to \"(.*?)\" the parameters:$" where proxy issue was handled. so I thought will go with "^I send put to \"(.*?)\" the parameters:$" & "^I send delete to \"(.*?)\" the parameters:$"

I was thinking to handle all these 3 methods post, put and delete in one but in existing code found that "The Cyclomatic Complexity of this method ".

hence separated the methods